### DISCUSSION TITLE
How to use custom messages in tools outside bazel ecosystem?

### DISCUSSION THREAD

**Discussion by thomasegriffith** [mvukov/rules_ros2#50] [General]
How does one make use of custom messages within tools that are outside of the bazel ecosystem?   Say I want to view a custom message in rviz or rqt or the ros-installed `ros2 topic echo`... is that possible?  Thanks!!

**Comment by mvukov** [comment]
Hi! Off the top of my head, I think you would need somehow to export (symlink?) custom msgs from your Bazel workspace to a colcon workspace and handle them in colcon-standard way.

**Reply by thomasegriffith** [reply]
That makes sense.  Thanks!

**Reply by thomasegriffith** [reply]
For vis: I did the steps below in a bash script, and now both `ros2 topic echo` and `ros2 bag record` work without issue.

1. copy all msg and srv files to some folder in `/tmp`
2. copy a `package.xml` and `CMakeLists.txt` into that folder (I used the ones for std_msgs as a template; just changed the package name)
3. cd into the new folder
4. colcon build it
5. source the `install/setup.bash` from the temp workspace
